### PR TITLE
Integrated nl api with demo site to visual results within map

### PR DESCRIPTION
**Pull request by TheRBajaj**
Integration between NL api and demo site. This feature allows you to search for places using the NL api upon toggle of the feature.
- added env variables for apiKey and url

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/atomicui/organisms/SearchBox/SearchBox.tsx [hunk 1]
```diff
@@ -104,12 +104,13 @@ const SearchBox: React.FC<SearchBoxProps> = ({
...
 					TriggeredByEnum.PLACES_SEARCH,
-					action
+					action,
+					isNLChecked
 				);
 				setIsSearching(false);
 			}, 200);
 		},
-		[mapRef, setIsSearching, search]
+		[mapRef, setIsSearching, search, isNLChecked]
```
Comment (thehoneymad) [inline src/atomicui/organisms/SearchBox/SearchBox.tsx:113]: Don't see how this is getting translated to `isNLSearchEnabled` in `search` method. What am I mising? :\

File: src/services/useAwsPlaceService.ts [hunk 2]
```diff
@@ -83,6 +84,15 @@ const useAwsPlaceService = () => {
...
 				return await locationClient?.searchPlaceIndexForPosition(params).promise();
+			},
+			getNLPlacesByText: async (Text: string) => {
+				const response = await fetch(`${NL_BASE_URL}/places/ask?` + new URLSearchParams({ Text: Text }), {
+					method: "GET",
+					headers: {
+						"x-api-key": NL_API_KEY
+					}
+				});
+				return await response.json();
```
Comment (thehoneymad) [inline src/services/useAwsPlaceService.ts:95]: Also, why do we also need to `await` this if we `await` the fetch in like 89.

**Review by thehoneymad — CHANGES_REQUESTED** [review]

File: index.html [hunk 3]
```diff
@@ -23,7 +23,7 @@
...
...onaws.com https://geofencing.geo.sa-east-1.amazonaws.com https://*.auth.us-east-1.amazoncognito.com https://*.auth.us-east-2.amazoncognito.com https://*.auth.us-west-2.amazoncognito.com https://*.auth.ap-southeast-1.amazoncognito.com https://*.auth.ap-southeast-2.amazoncognito.com https://*.auth.ap-northeast-1.amazoncognito.com https://*.auth.ca-central-1.amazoncognito.com https://*.auth.eu-central-1.amazoncognito.com https://*.auth.eu-west-1.amazoncognito.com https://*.auth.eu-west-2.amazoncognito.com https://*.auth.eu-north-1.amazoncognito.com https://*.auth.sa-east-1.amazoncognito.com https://cognito-idp.us-east-1.amazonaws.com https://cognito-idp.us-east-2.amazonaws.com https://cognito-idp.us-west-2.amazonaws.com https://cognito-idp.ap-southeast-1.amazonaws.com https://cognito-idp.ap-southeast-2.amazonaws.com https://cognito-idp.ap-northeast-1.amazonaws.com https://cognito-idp.ca-central-1.amazonaws.com https://cognito-idp.eu-central-1.amazonaws.com https://cognito-idp.eu-west-1.amazonaws.com https://cognito-idp.eu-west-2.amazonaws.com https://cognito-idp.eu-north-1.amazonaws.com https://cognito-idp.sa-east-1.amazonaws.com https://h57jdz559a.execute-api.us-west-2.amazonaws.com"
```
Comment (marinrelatic) [inline index.html:26]: Can we use NL_BASE_URL here as well?

File: src/core/constants/appConfig.ts [hunk 4]
```diff
@@ -63,7 +63,9 @@ const appConfig = {
 		APPLE_APP_STORE_LINK: getEnv("VITE_APPLE_APP_STORE_LINK"),
 		GOOGLE_PLAY_STORE_LINK: getEnv("VITE_GOOGLE_PLAY_STORE_LINK"),
 		COUNTRY_EVALUATION_URL: import.meta.env.DEV ? "/doNotDelete.txt" : "assets/doNotDelete.txt",
-		APP_VERSION: getEnv("VITE_APP_VERSION")
+		APP_VERSION: getEnv("VITE_APP_VERSION"),
+		NL_BASE_URL: getEnv("VITE_NL_BASE_URL"),
+		NL_API_KEY: getEnv("VITE_NL_API_KEY")
```
Comment (marinrelatic) [inline src/core/constants/appConfig.ts:68]: We should update README and .env.example

**Review by marinrelatic — CHANGES_REQUESTED** [review]

File: src/core/constants/appConfig.ts (hunk 4 above)
Comment (TheRBajaj) [reply src/core/constants/appConfig.ts:68]: Shouldn't we update that once we push it to mainline when everyone can access the changes?

File: index.html (hunk 3 above)
Comment (TheRBajaj) [reply index.html:26]: Ive tried to use the env variables to enter the url but it doesn't seem to work. Ive also tried some solutions online but no luck.

File: src/core/constants/appConfig.ts (hunk 4 above)
Comment (marinrelatic) [reply src/core/constants/appConfig.ts:68]: We should update it as part of this change, but keep it under the same branch

File: README.md
Comment (TheRBajaj) [inline README.md:1]: updated README.md and env.example file. Will update on how to get these values later!

**Review by marinrelatic — APPROVED** [review]

**Pull request merged**